frontend/lib/resolution/disambiguation.cpp

```diff
@@ -328,6 +328,18 @@ findMostSpecificCandidatesQuery(ResolutionContext* rc,
   CHPL_RESOLUTION_QUERY_BEGIN(findMostSpecificCandidatesQuery, rc,
               lst, call, callInScope, callInPoiScope);
 
+  if (call.name() == "this") {
+    if (call.actual(0).type().type() && call.actual(0).type().type()->isArrayType() &&
+        call.actual(0)
+            .type().type()
+            ->toArrayType()
+            ->eltType()
+            .type()
+            ->isStringType()) {
+      gdbShouldBreakHere();
+    }
+  }
+
```

Comment:
ready for review? Not so 😄 

I see you're taking a page from my playbook here.